modules/product/src/main/java/com/opengamma/strata/product/bond/BillTrade.java

```diff
@@ -0,0 +1,557 @@
+/*
+ * Copyright (C) 2018 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.bond;
+
+import java.io.Serializable;
+
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.gen.PropertyDefinition;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.PortfolioItemSummary;
+import com.opengamma.strata.product.ProductType;
+import com.opengamma.strata.product.ResolvableTrade;
+import com.opengamma.strata.product.SecuritizedProductTrade;
+import com.opengamma.strata.product.TradeInfo;
+import com.opengamma.strata.product.common.SummarizerUtils;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+import org.joda.beans.Bean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+/**
+ * A trade representing a bill.
+ * <p>
+ * A trade in an underlying {@link FixedCouponBond}.
+ * 
+ * <h4>Price</h4>
+ * Strata uses <i>decimal prices</i> for bonds in the trade model, pricers and market data.
+ * For yield, a price of 1.32% is represented in Strata by 0.0132.
+ */
+@BeanDefinition(constructorScope = "package")
+public final class BillTrade
+    implements SecuritizedProductTrade<Bill>, ResolvableTrade<ResolvedBillTrade>, ImmutableBean, Serializable {
+
+  /**
+   * The additional trade information, defaulted to an empty instance.
+   * <p>
+   * This allows additional information to be attached to the trade.
+   */
+  @PropertyDefinition(validate = "validateSettlement", overrideGet = true)
+  private final TradeInfo info;
+  /**
+   * The bill that was traded.
+   * <p>
+   * The product captures the contracted financial details of the trade.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final Bill product;
+  /**
+   * The quantity that was traded.
+   * <p>
+   * This will be positive if buying and negative if selling.
+   */
+  @PropertyDefinition(overrideGet = true)
+  private final double quantity;
+  /**
+   * The <i>yield</i> or <i>rate</i> at which the bill was traded, in decimal form.
+   * <p>
+   * Strata uses <i>decimal rates</i> for bill in the trade model, pricers and market data.
+   * For example, a yield of 1.32% is represented in Strata by 0.0132.
+   */
+  @PropertyDefinition
+  private final double yield;
```

Comment:
Given that we have the field as `price` for bonds, I think we should do the same here, and just document that it means "yield"